mkosi/config.py

```diff
@@ -1555,69 +1617,66 @@ def __init__(self) -> None:
         self.settings_lookup_by_dest = {s.dest: s for s in self.SETTINGS}
         self.match_lookup = {m.name: m for m in self.MATCHES}
 
-    def parse_config(self, path: Path, namespace: argparse.Namespace, defaults: argparse.Namespace) -> bool:
-        extras = path.is_dir()
+    def match_config(self, path: Path, namespace: argparse.Namespace, defaults: argparse.Namespace) -> bool:
+        triggered = None
 
-        if path.is_dir():
-            path = path / "mkosi.conf"
+        if not path.exists():
+            return True
 
-        parser = configparser.ConfigParser(
-            delimiters="=",
-            comment_prefixes="#",
-            inline_comment_prefixes="#",
-            empty_lines_in_values=True,
-            interpolation=None,
-            strict=False,
-            dict_type=ConfigParserMultipleValues,
-        )
+        for _, k, v in parse_ini(path, only_sections=["Match"]):
+            trigger = v.startswith("|")
+            v = v.removeprefix("|")
+            negate = v.startswith("!")
+            v = v.removeprefix("!")
 
-        parser.optionxform = lambda optionstr: optionstr # type: ignore
+            if not v:
+                die("Match value cannot be empty")
 
-        if path.exists():
-            parser.read(path)
+            if (s := self.settings_lookup_by_name.get(k)):
+                if not (match := s.match):
```

Comment:
here I'd actually not use the walrus operator, since the use of match is a bit down with something in between and I had to think to remember that they are the same.
```suggestion
                match = s.match
                if not match:
```